lib/std/heap.zig

```diff
@@ -8,6 +8,141 @@ const c = std.c;
 const Allocator = std.mem.Allocator;
 const windows = std.os.windows;
 
+const page_size_os: ?usize = if (builtin.os.tag.isDarwin())
+    switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 16 << 10,
+        else => null,
+    }
+else if (builtin.os.tag == .windows)
+    // -- <https://devblogs.microsoft.com/oldnewthing/20210510-00/?p=105200>
+    switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // SuperH => 4 << 10,
+        .mips, .mipsel, .mips64, .mips64el => 4 << 10,
+        .powerpc, .powerpcle, .powerpc64, .powerpc64le => 4 << 10,
+        // DEC Alpha => 8 << 10,
+        // Itanium => 8 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 4 << 10,
+        else => null,
+    }
+else
+    null;
```

Comment:
My conclusion from our last discussion on Discord is that it is fundamentally misguided to try to define page size based on architecture alone. Some architectures (e.g. LoongArch) allow some truly insane page size values that no OS is likely to ever allow.

So, really, what should happen here is that `page_size_os` should go away. `min_page_size` and `max_page_size` should switch on the OS, then the arch, and pick the appropriate value; any unhandled OS/arch should be a `@compileError()`. (I think it's ok if you can't find the right values for every OS we support; just make sure CI at least passes on the 3 main ones, and other people will be able to contribute the right values for less popular OSs.)

To be clear, by "appropriate value", I mean:

* `min_page_size`: The lowest page alignment requirement that `mmap` and friends can be configured to have.
* `max_page_size`: The highest page alignment requirement that `mmap` and friends can be configured to have.

Reply:
I'm not sure what you are trying to get at with this proposal.
Is it just that you want all targets to fail until they are explicitly documented? It feels rather extreme given the overheads involved. Like, GPA will have 30% more overhead than ARM in the extreme case of Hexagon, and even that might be necessary for portability.

Reply:
That is a good reminder though. FreeBSD seems to have one fixed page size as well, so it might be pretty easy to document all the Tier 2 supported targets.

Reply:
```zig
pub const min_page_size = switch (os) {
    .linux => switch (arch) {
        .hexagon => ...,
        .x86, x86_64 => ...,
        // ... etc ...
        else => @compileError(),
    }
    // ... etc ...
    else => @compileError(),
};
```

And likewise for `max_page_size`. I am saying that there should never be a fallback value that you pick based on architecture, absent OS. This is because it is ultimately the OS that dictates what page size values are possible, even if the architecture technically supports way bigger page sizes. For example, I think LoongArch supports page sizes up to 4G or something extremely silly like that, but Linux is never going to support that as the regular page size.